src/OpenTelemetry.Exporter.ZPages/Implementation/ZPagesSpans.cs

```diff
@@ -0,0 +1,123 @@
+{
+    /// <summary>
+    /// Implements the ZPages Span Queue which stores all the required span information.
+    /// </summary>
+    public static class ZPagesSpans
+    {
+        private static long startTime;
+
+        /// <summary>
+        /// Initializes static members of the <see cref="ZPagesSpans"/> class.
+        /// </summary>
+        static ZPagesSpans()
+        {
+            ZQueue = new LinkedList<Dictionary<string, ZPagesSpanInformation>>();
+            ProcessingSpanList = new Dictionary<string, long>();
+            CurrentMinuteSpanList = new ConcurrentDictionary<string, ZPagesSpanInformation>();
+            CurrentHourSpanList = new ConcurrentDictionary<string, ZPagesSpanInformation>();
+            TotalSpanCount = new Dictionary<string, long>();
+            TotalEndedSpanCount = new Dictionary<string, long>();
+            TotalSpanErrorCount = new Dictionary<string, long>();
+            TotalSpanLatency = new Dictionary<string, long>();
+            startTime = DateTimeOffset.Now.ToUnixTimeMilliseconds();
+        }
+
+        /// <summary>
+        /// Gets or sets ZQueue which stores the minute-wise span information.
+        /// </summary>
+        public static LinkedList<Dictionary<string, ZPagesSpanInformation>> ZQueue { get; set; }
+
+        /// <summary>
+        /// Gets or sets the current minute span information list.
+        /// </summary>
+        public static ConcurrentDictionary<string, ZPagesSpanInformation> CurrentMinuteSpanList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the current hour span information list.
+        /// </summary>
+        public static ConcurrentDictionary<string, ZPagesSpanInformation> CurrentHourSpanList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the processing span information list. This holds the names of the spans which have not ended yet, along with the active count.
+        /// </summary>
+        public static Dictionary<string, long> ProcessingSpanList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the count of spans name-wise.
+        /// </summary>
+        public static Dictionary<string, long> TotalSpanCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the count of ended spans name-wise.
+        /// </summary>
+        public static Dictionary<string, long> TotalEndedSpanCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the count of span errors according to span name.
+        /// </summary>
+        public static Dictionary<string, long> TotalSpanErrorCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the total latency of spans name-wise.
+        /// </summary>
+        public static Dictionary<string, long> TotalSpanLatency { get; set; }
+
+        /// <summary>
+        /// Gets or sets the retention time (in milliseconds) for the metrics.
+        /// </summary>
+        public static long RetentionTime { get; set; }
```

Comment:
Maybe this could be a `TimeSpan` instead of long? That will allow you set it using TimeSpan.FromXXX utility functions (eg `TimeSpan.FromSeconds(60)`) and time comparisons do not need to be converted using `ToUnixTimeMilliseconds`.